dynawo/sources/Modeler/Common/DYNSubModelFactory.cpp

```diff
@@ -49,19 +49,31 @@ SubModelFactories::~SubModelFactories() {
 }
 
 SubModelFactories::SubmodelFactoryIterator SubModelFactories::find(const std::string& lib) {
+#ifdef LANG_CXX11
+  std::unique_lock<std::mutex> lock(factoriesMutex_);
+#endif
   return (factoryMap_.find(lib));
 }
 
 bool SubModelFactories::end(SubmodelFactoryIterator& iter) {
+#ifdef LANG_CXX11
+  std::unique_lock<std::mutex> lock(factoriesMutex_);
```

Comment:
why a mutex in a  read only method?

Reply:
In case we perform a get and a set at the same time